cmd/podman/tree.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containers/libpod/cmd/podman/cliconfig"
+	"github.com/containers/libpod/cmd/podman/libpodruntime"
+	"github.com/containers/libpod/libpod/image"
+	units "github.com/docker/go-units"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	treeCommand cliconfig.TreeValues
+
+	treeDescription = "Prints layer hierarchy of an image in a tree format"
+	_treeCommand    = &cobra.Command{
+		Use:   "tree",
+		Short: treeDescription,
+		Long:  treeDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			treeCommand.InputArgs = args
+			treeCommand.GlobalFlags = MainGlobalOpts
+			return treeCmd(&treeCommand)
+		},
+		Example: "podman image tree alpine:latest",
+	}
+)
+
+func init() {
+	treeCommand.Command = _treeCommand
+	treeCommand.SetUsageTemplate(UsageTemplate())
+	treeCommand.Flags().Bool("whatrequires", false, "Show all child images and layers of the specified image")
+}
+
+// infoImage keep information of Image along with all associated layers
+type infoImage struct {
+	// id of image
+	id string
+	// tags of image
+	tags []string
+	// layers stores all layers of image.
+	layers []image.LayerInfo
+}
+
+func treeCmd(c *cliconfig.TreeValues) error {
+	args := c.InputArgs
+	if len(args) == 0 {
+		return errors.Errorf("an image name must be specified")
+	}
+	if len(args) > 1 {
+		return errors.Errorf("you must provide at most 1 argument")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	image, err := runtime.ImageRuntime().NewFromLocal(args[0])
+	if err != nil {
+		return err
+	}
+
+	// Fetch map of image-layers, which is used for printing output.
+	layerInfoMap, err := image.GetLayersMapWithImageInfo()
+	if err != nil {
+		return errors.Wrapf(err, "error while retriving layers of image %q", image.InputName)
+	}
+
+	// Create an imageInfo and fill the image and layer info
+	imageInfo := &infoImage{
+		id:   image.ID(),
+		tags: image.Names(),
+	}
+
+	size, err := image.Size(context.Background())
+	if err != nil {
+		return errors.Wrapf(err, "error while retriving image size")
+	}
+	fmt.Printf("Image ID: %s\n", imageInfo.id[:12])
+	fmt.Printf("Tags\t: %s\n", imageInfo.tags)
+	fmt.Printf("Size\t: %v\n", units.HumanSizeWithPrecision(float64(*size), 4))
```

Comment:
Writing the colon immediately after the label (`Tags:\t` etc.) is more idiomatic in English I think.